Hw29 spring actuator

demen777 opened the pull request:
Домашнее задание
Использовать метрики, healthchecks и logfile к приложению
Цель: Цель: реализовать production-grade мониторинг и прозрачность в приложении Результат: приложение с применением Spring Boot Actuator
Данное задание выполняется на основе одного из реализованных Web-приложений


1. Подключить Spring Boot Actuator в приложение метрики.
2. Включить метрики, healthchecks и logfile.
3. Реализовать свой собственный HealthCheck индикатор
4. UI для данных от Spring Boot Actuator реализовывать не нужно.
5. Опционально: переписать приложение на HATEOAS принципах с помощью Spring Data REST Repository

demen777 committed:
Add actuator dependency and properties

Commit:
Add custom health check

Commit:
Define spring.data.rest.base-path property

stvort commented on hw29_modern_web_spring_actuator/src/main/java/ru/otus/demen/books/actuator/ActuatorConfiguration.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class ActuatorConfiguration {
+    @Bean
+    public BooksHealthIndicator booksHealthIndicator(
+        RestTemplateBuilder restTemplateBuilder,
+        @Value("${server.port}") int httpPort,
+        @Value("${actuator.health.books.urlForCheck}") String urlForCheck)
+    {
+        String fullUrl = String.format("http://localhost:%d%s", httpPort, urlForCheck);
+        return new BooksHealthIndicator(restTemplateBuilder.build(), fullUrl);
+    }
+}
```
А точно это необходимо? Вроде это прямо в компонент внедрить можно. Ну и лучше url (хост) не хардкодить)

demen777 replied:
Перенес загрузку значений из properties в конструктор компонента
Добавил property для hostname

stvort commented on hw29_modern_web_spring_actuator/src/main/resources/application.yml:
```diff
@@ -0,0 +1,44 @@
+  endpoints:
+    web:
+      exposure:
+        include: health,loggers,metrics
```
Ага. Только не loggers, а logfile

demen777 replied:
Пофиксил

demen777 committed:
Change loggers to logfile
Remove configuration from ActuatorConfiguration.java to BooksHealthIndicator
Add actuator.health.books.host-for-check property
Rename actuator.health.books.urlForCheck to path-for-check
